### PR TITLE
SPU Analyzer: Fix programs code caching

### PR DESCRIPTION
**Pull request by elad335**
Was treating programs which are derivatives of others as equals.

Test with #16310

### CODE REVIEW

File: rpcs3/Emu/Cell/SPUCommonRecompiler.cpp [hunk 1]
```diff
@@ -1264,7 +1264,22 @@ bool spu_program::operator<(const spu_program& rhs) const noexcept
...
+
+	int cmp0 = 0;
+
+	if (lhs_data.size() != rhs_data.size())
+	{
+		cmp0 = std::memcmp(lhs_data.data(), rhs_data.data(), std::min(lhs_data.size_bytes(), rhs_data.size_bytes()));
+
+		if (cmp0 == 0 && lhs_data.size() < rhs_data.size())
+		{
+			cmp0 = lhs_data.size() < rhs_data.size() ? -1 : 1;
```
Comment (MSuih) [inline rpcs3/Emu/Cell/SPUCommonRecompiler.cpp:1276]: Isn't result of `lhs_data.size() < rhs_data.size()` already known at this point?

Comment (elad335) [reply rpcs3/Emu/Cell/SPUCommonRecompiler.cpp]: Fixed

Comment (MSuih) [reply rpcs3/Emu/Cell/SPUCommonRecompiler.cpp]: The same issue is on line 45 now

Comment (elad335) [reply rpcs3/Emu/Cell/SPUCommonRecompiler.cpp]: Its outdated

Comment (MSuih) [reply rpcs3/Emu/Cell/SPUCommonRecompiler.cpp]: Ah yeah, I was one force push behind

**Comment by boti0719** [comment]
Sorry for texting here but i got a problem with todays update:
I played god of war III yesterday and its run close perfetly but after this update the game dosen't even have one fps and i hope that i highlighted some problem you can fix.

**Pull request merged**